lib/src/connector/course_connector.dart

```diff
@@ -101,38 +101,41 @@ class CourseConnector {
       //取得學期資料
       nodes = courseNodes[0].getElementsByTagName("td");
       SemesterJson semester = SemesterJson();
-      semester.year = nodes[1].text;
-      semester.semester = nodes[2].text;
+
+      final titleString = nodes[0].text;
+      RegExp regex = RegExp(r'\b\d+\b');
+      Iterable<RegExpMatch> matches = regex.allMatches(titleString);
+      List<String> numbers = matches.map((match) => match.group(0)).toList();
```

Comment:
```suggestion
      RegExp regex = RegExp(r'\b\d+\b');
      Iterable<RegExpMatch> matches = regex.allMatches(titleString);
      List<String> numbers = matches.map((match) => match.group(0)).toList();
      
      if (numbers.isEmpty) {
        // do something...
      }
      if (numbers.length < 3) {
        // do something...
      }
```

Suggest adding null check before using content inside `numbers`.
It is because `Match.group` returns `String?`.
```dart
// pattern.dart:109
String? group(int group);
```

Reply:
should i issue an error directly when the numbers filter isn't working properly?

Reply:
As the `try-catch` structure can avoid error/exception be thrown to upper layer, it is not design only for this particular error case. Furthermore, the potential exception here is `IndexOutOfRange` error, which is really hard to be identified in the future debugging work.
So here I'll suggest using custom log message or custom exception with specific message to clearly show what's wrong.

Reply:
ok but is there a standard format of exception message that TAT is using or i can just use throw() to throw a general issue

Reply:
Unfortunately, there's no any rules about error throwing in TAT.
However, TAT-core does.
It currently looks like: `[TAT core] <namespace> <position descriptor>`

Reply:
ok

Reply:
sorry i might need more details, should i do it simply like this: 
```
      if (studentSemesterDetails.isEmpty) {
        throw RangeError("[TAT flutter] course_connector.dart: studentSemesterDetails list is empty");
      }

```
or should i create an addition class of this so we can generally use them in the future

Reply:
I think this format is okay, you could even abbreviate it to `[TAT] ...`.
Regarding how to handle the situation when the list is empty or has less than three elements, whether to terminate the operation (throw an error) or to continue running but take some measures, I am currently having a hard time helping, please make your own judgment.
As for where the final error is thrown, whether it should be thrown, etc., I might not be able to give you clear instructions, because I haven't actually studied whether your changes can actually fix the problem. _(I've already stepped back, so I might not have time to deal with these things, sorry about that ...)_

Reply:
ok, considering this patch is for temparary result for the current issues due to the policy changes of NTUT and the reconstruction of TAT that we are going to make, i will make this simple for now and leave the details to the pr in the future

Reply:
[094f8c5](https://github.com/NEO-TAT/tat_flutter/pull/214/commits/094f8c525e7a31a321a52e25026ab767ae4dc00f)